qiskit_machine_learning/circuit/library/raw_feature_vector.py

```diff
@@ -164,7 +164,7 @@ def _define(self):
         # cast ParameterExpressions that are fully bound to numbers
         cleaned_params = []
         for param in self.params:
-            if len(param.parameters) == 0:
+            if isinstance(param, (float, complex)) or len(param.parameters) == 0:
```

Comment:
I'm wondering if this could cause issues if `numpy` datatypes are used. In that case, we could also just check if (a) the parameter is _not_ a parameter expression, or (b) if it is parameter expression with 0 parameters.
```suggestion
            if not isinstance(param, ParameterExpression) or len(param.parameters) == 0:
```

Reply:
It makes sense. Thanks.